templates/templates.go

```diff
@@ -0,0 +1,136 @@
+package templates
+
+import (
+	"embed"
+	"fmt"
+	permission1 "github.com/ystv/web-auth/infrastructure/permission"
+	"github.com/ystv/web-auth/permission"
+	"github.com/ystv/web-auth/permission/permissions"
+	"github.com/ystv/web-auth/role"
+	"github.com/ystv/web-auth/user"
+	"html/template"
+	"io"
+	"log"
+	"time"
+)
+
+//go:embed *.tmpl
+var tmpls embed.FS
+
+type (
+	Templater struct {
+		Permission *permission.Store
+		Role       *role.Store
+		User       *user.Store
+	}
+
+	Template     string
+	TemplateType int
+)
+
+const (
+	ForgotTemplate       Template = "forgot.tmpl"
+	NotFound404Template  Template = "404NotFound.tmpl"
+	ForgotEmailTemplate  Template = "forgotEmail.tmpl" // generated by go generate
+	InternalTemplate     Template = "internal.tmpl"
+	LoginTemplate        Template = "login.tmpl"
+	NotificationTemplate Template = "notification.tmpl"
+	ResetTemplate        Template = "reset.tmpl"
+	ErrorTemplate        Template = "error.tmpl"
+	SettingsTemplate     Template = "settings.tmpl"
+	SignupTemplate       Template = "signup.tmpl"
+	UserTemplate         Template = "user.tmpl"
+	UsersTemplate        Template = "users.tmpl"
+	RolesTemplate        Template = "roles.tmpl"
+	RoleTemplate         Template = "role.tmpl"
+	ResetEmailTemplate   Template = "resetEmail.tmpl" // generated by go generate
+	PermissionsTemplate  Template = "permissions.tmpl"
+	PermissionTemplate   Template = "permission.tmpl"
+
+	NoNavType      TemplateType = 0
+	PaginationType TemplateType = 1
+	RegularType    TemplateType = 2
+)
+
```

Comment:
SUPER DUPER NITPICK: Personally I'd have done this like

```go
type Template string
const (
	ForgotTemplate       Template = "forgot.tmpl"
	NotFound404Template  Template = "404NotFound.tmpl"
	// ...
)

type TemplateType int
const (
	NoNavType TemplateType = iota
	PaginationType TemplateType
	RegularType TemplateType
)
```

(grouping together all the constants for each type with the type declaration - right now I have to scroll a ways up to see what `TemplateType` is). But I'm happy for this to stay as it is.